src/SonarScanner.MSBuild.PreProcessor/WebClientDownloaderBuilder.cs

```diff
@@ -177,10 +180,23 @@ private static bool ServerCertificateValidationChain(X509Certificate2Collection
             var rootInChain = testChain.ChainElements.Cast<X509ChainElement>().Last();
             var foundInTrustStore = trustStore.Find(X509FindType.FindBySerialNumber, rootInChain.Certificate.SerialNumber, validOnly: false);
             // Check if the certificates found by serial number in the trust store really contain the root certificate of the chain by doing a proper equality check
-            return IsCertificateInTrustStore(rootInChain.Certificate, foundInTrustStore);
+            if (IsCertificateInTrustStore(rootInChain.Certificate, foundInTrustStore))
+            {
+                return true;
+            }
+            else
+            {
+                logger.LogWarning(Resources.WARN_TrustStore_Chain_RootCertificateNotFound,
+                                  rootInChain.Certificate.Issuer,
+                                  rootInChain.Certificate.Thumbprint,
+                                  trustStoreFile,
+                                  SonarProperties.TruststorePath);
+                return false;
```

Comment:
This code path is untestable and likely unreachable (because it requires a exclusively "partial chain" error when the callback is invoked and a sub-sequent exclusively "untrusted root" error here which can only be achieved if the root ca is in the truststore file).

I would still keep the code path as is, just to be safe. If we ever encounter this in some logs, we can try to figure out the environment (most likely some OS specific scenario) in which this could happen.